packages/teuchos/parameterlist/test/ParameterList/ParameterList_UnitTests.cpp

```diff
@@ -1208,6 +1208,48 @@ TEUCHOS_UNIT_TEST( ParameterList, print ) {
   }
 }
 
+TEUCHOS_UNIT_TEST( ParameterList, NonPrintableParameterEntries){
+  // test printing std::vector<int> from a parameter list
+  {
+    ParameterList paramList = ParameterList("std::vector test");
+    std::vector<int> testVec = {1};
+    paramList.set("My std::vector<int>", testVec);
+
+    try {
+      paramList.print();  // Should throw!
+      TEUCHOS_TEST_FOR_EXCEPTION(true, std::logic_error, "If you get here then the test failed!");
+    }
+    catch (const NonprintableParameterEntryException &except) {
+      std::string actualMessage = except.what();
+      std::string expectedMessage = "Trying to print type std::vector<int, std::allocator<int> > which is not printable (i.e. does not have operator<<() defined)!";
+      TEST_ASSERT(actualMessage.find(expectedMessage) != std::string::npos);
+    }
+  }
+
+  // test printing enum class from a parameter list
+  {
+    ParameterList paramList = ParameterList("enum class test");
+    enum class Shape : int { CIRCLE, SQUARE, TRIANGLE };
+    paramList.set("My enum class", Shape::SQUARE);
+
+    try {
+      paramList.print();  // Should throw!
+      TEUCHOS_TEST_FOR_EXCEPTION( true, std::logic_error, "If you get here then the test failed!" );
+    }
+    catch (const NonprintableParameterEntryException &except) {
+      std::string actualMessage = except.what();
+      std::string expectedMessage =
+              "Trying to print type "
+              "Teuchos::ParameterList_NonPrintableParameterEntries_UnitTest::runUnitTestImpl(Teuchos::basic_FancyOStream<char, std::char_traits<char> >&, bool&) const::Shape "
+              "which is not printable (i.e. does not have operator<<() defined)!";
```

Comment:
That is a long and interesting type name:

`Teuchos::ParameterList_NonPrintableParameterEntries_UnitTest::runUnitTestImpl(Teuchos::basic_FancyOStream<char, std::char_traits<char> >&, bool&) const::Shape`

😊

The name of the function `runUnitTestImpl()` is an implementation detail that we don't want to expose in a general test.

Also, I wonder if that name will be portable across all compilers?  I wonder if it might be more portable to declare this type `enum class Shape` in an anonymous namespace in this file `PameterList_UnitTests.cpp` and then use that in this test?

Reply:
I have moved it to an anonymous namespace. Done.